app/src/main/java/br/com/solutis/viciadosemfilmes/DetalheFilmeActivity.java

```diff
@@ -0,0 +1,160 @@
+package br.com.solutis.viciadosemfilmes;
+
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.ProgressBar;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.constraintlayout.widget.ConstraintLayout;
+
+import com.bumptech.glide.Glide;
+
+import java.io.IOException;
+import java.util.Calendar;
+
+import br.com.solutis.viciadosemfilmes.config.RetrofitConfig;
+import br.com.solutis.viciadosemfilmes.model.Filme;
+import br.com.solutis.viciadosemfilmes.model.Genero;
+import br.com.solutis.viciadosemfilmes.service.FilmeService;
+import br.com.solutis.viciadosemfilmes.ui.constants.Constantes;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class DetalheFilmeActivity extends AppCompatActivity {
+
+    private long id;
+    private ImageView capa;
+    private TextView titulo;
+    private TextView ano;
+    private TextView tagline;
+    private TextView genero;
+    private TextView sinopse;
+    private ImageView favedIcon;
+    private ImageView unFavedIcon;
+    private SharedPreferences preferences;
+    private ProgressBar progressBar;
+    private ConstraintLayout telaDetalhe;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_detalhe_filme);
+        Intent dados = getIntent();
+        preferences = getSharedPreferences(Constantes.USER_PREFERENCES, MODE_PRIVATE);
+        carregaIdFilme(dados);
+
+        try {
+            consultaFilme();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void carregaIdFilme(Intent dados) {
+        if(dados.hasExtra(Constantes.EXTRA_KEY)){
```

Comment:
Não tem tratamento de erro, sempre que o ID não for encontrado o app carregara o favorito, mas se ocorrer algum comportamento anomalo e o extra não for passado?